pkg/ibmcsidriver/controller.go

```diff
@@ -127,6 +127,118 @@ func (csiCS *CSIControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
+		the zone provided by user as used by the volume (aka File share).In this case any IP adress associated with PrimaryIPID is assigned to VNI.
+
+		Case 5: User has not provided the subnetId and provided zone, PrimaryIPID
+		The VolumeAccessPoint (aka File share target) will be created with mountPath having IP adress associated with PrimaryIPID from subnet within
+		the zone provided by user as used by the volume (aka File share).In this case any IP adress associated with PrimaryIPID is assigned to VNI.
+
+		Case 6: User has not provided the subnetId,zone but provided PrimaryIPID
+		This will throw error that zone is mandatory as CSI driver cannot predict the zone in such scenarios CSI will pick this up from topology.
+
+		Case 7: User has provided the subnetId but nothing else
+		This will throw error that zone is mandatory as CSI driver cannot predict the zone in such scenarios CSI will pick this up from topology.
+
+		Case 8: User has not provided the subnetID but provided the zone and PrimaryIPAddress
+		This will throw error that subnet is mandatory as CSI driver cannot predict the subnet in such scenarios as there
+		might be multiple subnets in same zone.
+
+		Case 9: User has provided the subnetID and PrimaryIPAddress but not provided the zone.
+		This will throw error that zone is mandatory as CSI driver cannot predict the zone in such scenarios CSI will pick this up from topology.
+
+		In all the above cases the variation possible is user can pass 0 or more securitygroupIDs that will govern the authorization. IF user does not pass
+		any securityGroupID then
+
+		1. IKS cluster security group is fetched and used by CSI driver
+		2. Else default VPC security group is considerd by the VPC IAAS layer.
+
+	*/
+
+	if requestedVolume.AccessControlMode == SecurityGroup {
+		/* Skip GetSubnetForVolumeAccessPoint call if user has not provided SubnetID but PrimaryIPID is provided.
+		For all rest of the following use cases if subnetId is not provided we fetch subnet
+		1.) User has not provided anything just ENI/VNI is enabled (Any random IP Address will be created and assigned to VNI in the fetch subnet range)
+		2.) User has provided PrimaryIP Address. (The respective IP Address will be created and assigned to VNI in the fetched subnet range)
+		*/
+		subnetID := requestedVolume.SubnetID
+
+		if len(subnetID) == 0 && (requestedVolume.PrimaryIP == nil || len(requestedVolume.PrimaryIP.ID) == 0) {
+			subnetIDList := os.Getenv("VPC_SUBNET_IDS")
+
+			//We need to abort here as there is no use of going ahead and fetching the matching subnet with empty list
+			if len(subnetIDList) == 0 {
+				return nil, commonError.GetCSIError(ctxLogger, commonError.SubnetIDListNotFound, requestID, nil)
+			}
+
+			subnetReq := provider.SubnetRequest{
+				SubnetIDList:  subnetIDList,
+				ZoneName:      requestedVolume.Az,
+				VPCID:         os.Getenv("VPC_ID"),
+				ResourceGroup: requestedVolume.ResourceGroup,
```

Comment:
If resource is not given it will consider the resource group set in the storage-secret-store
```

[VPC]
  g2_token_exchange_endpoint_url = "https://iam.bluemix.net"
  g2_riaas_endpoint_url = "https://us-east.iaas.cloud.ibm.com"
  g2_riaas_endpoint_private_url = "https://us-east.private.iaas.cloud.ibm.com"
  g2_resource_group_id = "c2bb5b449d924d4da3e9430eea8d57ed"
```